File: src/api-sdks/connection-api/clients/csharp/src/IdeaStatiCa.ConnectionApi/ConnectionApiServiceAttacher.cs
using IdeaStatiCa.Api.Common;
using System.Threading.Tasks;

namespace IdeaStatiCa.ConnectionApi
{
	/// <summary>
	/// Factory for creating instances of Connection API client that are connected to the specified REST API service
	/// </summary>
	public class ConnectionApiServiceAttacher : IApiServiceFactory<IConnectionApiClient>
	{
		string BaseUrl { get; set; }

		private readonly string clientApplication;
		private readonly string clientApplicationVersion;

		/// <summary>
		/// Constructor
		/// </summary>
		/// <param name="baseUrl"> URL of the REST API service</param>
		/// <param name="clientApplication">
		/// Name of the application making the calls, for example "NorsokChecker" - a constant of the
		/// build, so it belongs to the factory rather than to a single call. Every client this factory
		/// creates is reported under it, which is what lets usage be attributed to an integration at
		/// all; see <see cref="ClientApplicationIdentity"/>. Optional.
		/// </param>
		/// <param name="clientApplicationVersion">Version of that application. Optional.</param>
		public ConnectionApiServiceAttacher(string baseUrl, string clientApplication = null,
			string clientApplicationVersion = null)
		{
			this.BaseUrl = baseUrl;
			this.clientApplication = clientApplication;
			this.clientApplicationVersion = clientApplicationVersion;
		}

		/// <inheritdoc cref="IApiServiceFactory{T}.CreateApiClient"/>
		public async Task<IConnectionApiClient> CreateApiClient()
		{
			var client = new ConnectionApiClient(BaseUrl, clientApplication, clientApplicationVersion);
			await client.CreateAsync();
			return client;
		}
	}		
}

File: src/api-sdks/connection-api/clients/python/ideastatica_connection_api/connection_api_service_attacher.py
import logging
from typing import Optional
from ideastatica_connection_api.connection_api_client import ConnectionApiClient

logger = logging.getLogger(__name__)

class ConnectionApiServiceAttacher:
    def __init__(self, base_url: str, client_application: Optional[str] = None,
                 client_application_version: Optional[str] = None):
        """
        :param base_url: URL of the running REST API service.
        :param client_application: Name of the application making the calls, for example
            "NorsokChecker" - a constant of the release, so it belongs to the factory rather than to
            a single call. Every client this factory creates is reported under it, which is what
            lets usage be attributed to an integration at all. Optional.
        :param client_application_version: Version of that application. Optional.
        """
        self.base_url = base_url
        self.client_application = client_application
        self.client_application_version = client_application_version

    def create_api_client(self) -> ConnectionApiClient:
        logger.info(f"Creating client attached to {self.base_url}")
        return ConnectionApiClient(self.base_url, self.client_application,
                                   self.client_application_version)